lms/djangoapps/instructor/views/api.py

```diff
@@ -2654,8 +2710,31 @@ def spoc_gradebook(request, course_id):
         }
         for student in enrolled_students
     ]
+    return student_info, page_num, total_pages, next_offset, previous_offset
+
+
+# ---- Gradebook (shown to small courses only) ----
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def spoc_gradebook(request, course_id):
+    """
+    Show the gradebook for this course:
+    - Only shown for courses with enrollment < settings.FEATURES.get("MAX_ENROLLMENT_INSTR_BUTTONS")
+    - Only displayed to course staff
+    """
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```

Comment:
`SlashSeparatedCourseKey.from_deprecated_string ` deserializes a course
key. It is old syntax, the new syntax is `CourseKey.from_string`. You
should NOT be invoking `SlashSeparatedCourseKey`. It (ought to) raise a
deprecation warning; unfortunately, we've silenced those warnings as there
are too many of them.

Please read up on
https://github.com/edx/edx-platform/wiki/Opaque-Keys-(Locators)

On Wed, Nov 4, 2015 at 2:43 PM, Giovanni Di Milia <notifications@github.com>
wrote:

> In lms/djangoapps/instructor/views/api.py
> <https://github.com/edx/edx-platform/pull/10250#discussion_r43928844>:
>
> > @@ -2654,8 +2710,31 @@ def spoc_gradebook(request, course_id):
> >          }
> >          for student in enrolled_students
> >      ]
> > +    return student_info, page_num, total_pages, next_offset, previous_offset
> > +
> > +
> > +# ---- Gradebook (shown to small courses only) ----
> > +@cache_control(no_cache=True, no_store=True, must_revalidate=True)
> > +@require_level('staff')
> > +def spoc_gradebook(request, course_id):
> > +    """
> > +    Show the gradebook for this course:
> > +    - Only shown for courses with enrollment < settings.FEATURES.get("MAX_ENROLLMENT_INSTR_BUTTONS")
> > +    - Only displayed to course staff
> > +    """
> > +    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
>
> Not sure what SlashSeparatedCourseKey.from_deprecated_string does, but in
> another PR we were told to do something like
>
> course_key = CourseKey.from_string(ccx_key)
>
> that I think works also for normal course (but I'm not sure)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10250/files#r43928844>.
>